FinancialTracker/Firebase.swift

```diff
@@ -66,20 +72,49 @@ class FirebaseHandler {
     }
 
     func signIn(email: String, password: String, completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
-        Auth.auth().signIn(withEmail: email, password: password) { _, error in
+        auth.signIn(withEmail: email, password: password) { _, error in
 
-            if error != nil {
+            guard error == nil else {
                 completionHandler(FirebaseError.auth(error), false)
+                return
             }
             completionHandler(nil, true)
         }
     }
     
-    func signOut() {
+    func signOut(completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
         do {
-            try Auth.auth().signOut()
+            try auth.signOut()
         } catch let signOutError {
-            print("Error signing out: %@", FirebaseError.signOut(signOutError))
+            completionHandler(FirebaseError.signOut(signOutError), false)
+        }
+    }
+    
+    func getCurrentUserData(completionHandler: @escaping (FirebaseError?, Bool, [String: Any]?) -> Void) {
```

Comment:
Looks like we need a user model too, just a function won't be enough it seems.

Reply:
Also, no need to pass in a `Bool` in the `completionHandler` if we are passing any other kind of data like the dictionary in the case.

So, you either have `(FirebaseError?, Bool)` or `(FirebaseError?, <Data of some kind>)`